Feature#10 frontend manage employee

AngelBFdev opened the pull request:
# Description
- Added new view for management employee
- Added modal to create a new employee
- Added styles for all the tables that will be use in other views since they share styles.

## Screenshots
Mockup Management employee
![image](https://user-images.githubusercontent.com/50933273/149381109-c01fef70-a06d-4c1f-ac26-35e2c3e49c93.png)
Frontend Management Employee
![image](https://user-images.githubusercontent.com/50933273/149381257-ee28b3a4-e937-4082-9f64-54f92952da42.png)
Mockup Add new Employee
![image](https://user-images.githubusercontent.com/50933273/149381345-7568e7c1-c74a-4241-9043-400eb2569a48.png)
Frontend Add new Employee
![image](https://user-images.githubusercontent.com/50933273/149381402-c016f3b1-9e6d-4e54-8de4-fe9e8ec3a1d9.png)

AngelBFdev committed:
Initialize branch
- Commented wdm gem in the Gemfile in order to work
- Added new route in config/routes.rb

AngelBFdev committed:
Added new page for management employee
- Added new action manage_employee in pages_controller.rb
- Created new page manage_employee.html.erb with the header of all the
  pages
- Added route /manage_employee pointing to pages#manage_employee

AngelBFdev committed:
Added Structure management employeee

- Added stylesheet table for future styles
- Generated manage_employee controller with action table
- Change pages_controller reset to base from branch master
- Change route to new controller, route now: "/manage_employee"

Co-authored-by: Villaedi <ediberto.sanchez.villalobos@gmail.com>

AngelBFdev committed:
Table base styles

- table.html.erb: Added class main, table, employe, addButton
- table.css.scss:
 - class table: Added background color, text align and margin
 - class addButton: Added text align right, background color, margin and sizes
   with paddin.
 - h1, class employe: added base styles

Co-authored-by: Villaedi <ediberto.sanchez.villalobos@gmail.com>

AngelBFdev committed:
Merge branch 'master' of https://github.com/BC-MAY-21-ROR/kata-07-reloj-checador-k7-n21-t3 into Feature#10Frontend-manage-employee

AngelBFdev committed:
Table Management Employee and Modal Add Employee design

- table.html.erb:
  - Added modal for new employee
  - Added example record employee in table
- table.css.scss:
  - Added styles for class employee, employee__title, employee__date and
    button. In order to give styles for the table
  - Added styles for class modal, form, form__label, form_select,
    form__button. In order to give styles to the modal for new employees

Co-authored-by: Villaedi <ediberto.sanchez.villalobos@gmail.com>
Co-authored-by: DerekAyala <derek.ayala12345@hotmail.com>

AngelBFdev committed:
Form modal design change

- table.html.erb: field position, employee, company and private number
  in the modal, now are in a div class col and row.
- table.css.scss: form__button width changed to 15rem and margin left
  now is 30rem.

ccarruitero commented on Gemfile:
```diff
@@ -3,7 +3,7 @@ git_source(:github) { |repo| "https://github.com/#{repo}.git" }
 ruby '3.0.2'
 
-gem 'wdm', '>= 0.1.0'
+# gem 'wdm', '>= 0.1.0'
```
I prefer to remove unused gems

AngelBFdev replied:
Another teammate need it in order to work with rails. But I had to comment it because it was giving me an error

ccarruitero reviewed (dismissed):
Seems good. Just an small comment

JuanequeX commented on app/views/manage_employee/table.html.erb:
```diff
@@ -0,0 +1,100 @@
+            <th class="employee__data">esanchez23@ucol.mx</th>
+            <th class="employee__data">Developer</th>
+            <th class="employee__data">Foo Corp</th>
+            <th class="employee__data"><button> <%= image_tag "edit.svg" =%></button> </th>
+            <th class="employee__data"><button><%= image_tag "disable.svg" =%></button></th>
```
To show a image the sintax requires only one equal sign, at this way `<%= image_tag "edit.svg" %>`

AngelBFdev replied:
Done

JuanequeX commented on app/assets/stylesheets/src/layout/table.css.scss:
```diff
@@ -0,0 +1,98 @@
+      text-align: center;
+    }
+    .form__button{
+      width: 15rem;
+      background-color: $background_table;
+      box-shadow: 0px 2px 1px 0px $secondary_color;
+      color: $font_color;
+      font-size: 1.5rem;
+      margin: 3rem 0rem 1rem 30rem;
+      padding: 1rem;
```
Sort this block

AngelBFdev replied:
Done

JuanequeX commented on app/assets/stylesheets/src/layout/table.css.scss:
```diff
@@ -0,0 +1,98 @@
+        border: 0.5rem solid $background_table;
+      }
+      .employee__data{
+        text-align: center;
+        background-color: $table_cell;
+        border: 0.5rem solid $background_table;
+        color:black;
+        font-size: 1.8rem;
```
Sort this block

AngelBFdev replied:
Done

JuanequeX reviewed (changes requested):
LGTM

YilKanda commented on app/views/manage_employee/table.html.erb:
```diff
@@ -0,0 +1,100 @@
+    <nav class="navbar">
+      <div class="container-fluid">
+        <span class="navbar-brand mb-0 h1" id="logo">
+            <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 134.976 29" width="144.976" height="39"><g fill="#FFF" color="#FFF" transform="translate(0, 0) scale(0.6041667)"><svg width="48.0" height="48.0" x="0.0" y="0.0" viewBox="0 0 48 48"><path fill="currentColor" fill-rule="nonzero" d="M24 .21L.21 24 24 47.79 47.79 24 24 .21zM5.82 24L24 5.82l7.69 7.69-2.8 2.8L24 11.43 11.43 24l4.88 4.89-2 ...
```
Can you put this SVG in an external file?

AngelBFdev replied:
Done

YilKanda commented on app/views/manage_employee/table.html.erb:
```diff
@@ -0,0 +1,100 @@
+        </span>
+
+        <div class="user_admin">
+          <svg width="22" height="22" viewBox="0 0 32 32" fill="none" xmlns="http://www.w3.org/2000/svg">
+            <path d="M16.0001 14.6667C18.9456 14.6667 21.3334 12.2789 21.3334 9.33333C21.3334 6.38781 18.9456 4 16.0001 4C13.0546 4 10.6667 6.38781 10.6667 9.33333C10.6667 12.2789 13.0546 14.6667 16.0001 14.6667Z" stroke="#00E8B0" stroke-width="2" stroke-linecap="round" stroke-linejoin="round"/>
+            <path d="M8 28V25.3333C8 23.9188 8.5619 22.5623 9.5621 21.5621C10.5623 20.5619 11.9188 20 13.3333 20H18.6667C20.0812 20 21.4377 20.5619 22.4379 21.5621C23.4381 22.5623 24 23.9188 24 25.3333V28" stroke="#00E8B0" stroke-width="2" stroke-linecap="round" stroke-linejoin="round"/>
+          </svg>
```
Can you put this SVG in an external file?

AngelBFdev replied:
Done

YilKanda commented on app/views/manage_employee/table.html.erb:
```diff
@@ -0,0 +1,100 @@
+                </div>
+              </div>
+            </div>
+          </div>
```
Can you put this modal in an external file?

AngelBFdev replied:
Done

AngelBFdev committed:
Modal with render

- table.css.scss: Sorted alphabetically
- table.html.erb: svg now are called by image_tag from ruby on rails,
  and now modal is called by render from ruby on rails
- shared/_modal.html.erb: html code for modal

JuanequeX reviewed (approved):
LGTM
